From 0a2f6ba03f488a69457b937ea7469cc2c65d66f9 Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Fri, 9 Feb 2018 12:46:08 +0100
Subject: [PATCH 23/25] ME176C: drm/i915: Avoid changing brightness to max
 after enable

Right now, the brightness is reset to maximum if it was set to
minimum before the backlight was disabled. This looks bad when
turning the screen back on because it will actually fade down
instead of fade up when a lower brightness is set.

Just keep the brightness as-is after turning backlight on so
userspace can directly set the correct brightness value.
---
 drivers/gpu/drm/i915/intel_panel.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 3b1c5d783ee7..8e0b177d6bba 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -1164,15 +1164,6 @@ void intel_panel_enable_backlight(const struct intel_crtc_state *crtc_state,
 
 	WARN_ON(panel->backlight.max == 0);
 
-	if (panel->backlight.level <= panel->backlight.min) {
-		panel->backlight.level = panel->backlight.max;
-		if (panel->backlight.device)
-			panel->backlight.device->props.brightness =
-				scale_hw_to_user(connector,
-						 panel->backlight.level,
-						 panel->backlight.device->props.max_brightness);
-	}
-
 	panel->backlight.enable(crtc_state, conn_state);
 	panel->backlight.enabled = true;
 	if (panel->backlight.device)
-- 
2.17.0

